tabular/src/autogluon/tabular/models/tab_transformer/pretexts.py

```diff
@@ -0,0 +1,151 @@
+from copy import deepcopy
+
+import numpy as np
+import torch
+import torch.nn as nn
+
+from ...constants import REGRESSION
+
+"""
+possible TODO: although there is a supervised pretext option below, i.e. pretrain using
+labeled data presumably from a different but related task, curently this cannot be handled 
+by the AG API. One simple way to handle it would be to have an optional "pretext label" 
+argument to task.fit that tells the pretraining module to use the supervised pretext task
+with that particular column as the label.
+"""
+
+
+class SUPERVISED_pretext(nn.Module):
+    # holder class to handle supervised pretraining.
+    def __init__(self, kwargs):
```

Comment:
why kwargs? Why not problem_type and device?

Reply:
Changed.